delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/acls/AclState.scala

```diff
@@ -0,0 +1,108 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model.acls
+
+import java.time.Instant
+
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.{nxv, schemas}
+import ch.epfl.bluebrain.nexus.delta.sdk.AclTargetResource
+import ch.epfl.bluebrain.nexus.delta.sdk.model.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourceRef.Latest
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Identity, ResourceF, ResourceRef}
+import org.apache.jena.iri.IRI
+
+/**
+  * Enumeration of ACLs states.
+  */
+sealed trait AclState extends Product with Serializable {
+
+  /**
+    * @return the current state revision
+    */
+  def rev: Long
+
+  /**
+    * @return the current deprecation status (always false for acls)
+    */
+  final def deprecated: Boolean = false
+
+  /**
+    * @return the schema reference that acls conforms to
+    */
+  final def schema: ResourceRef = Latest(schemas.acls)
+
+  /**
+    * @return the collection of known types of acls resources
+    */
+  final def types: Set[IRI] = Set(nxv.AccessControlList)
+
+  /**
+    * Converts the state into a resource representation.
+    *
+    * @param id the resource identifier
+    */
+  def toResource(id: IRI): AclTargetResource
+}
+
+object AclState {
+
+  /**
+    * Initial state type.
+    */
+  type Initial = Initial.type
+
+  /**
+    * Initial state for the permission set.
+    */
+  final case object Initial extends AclState {
+    override val rev: Long = 0L
+
+    override def toResource(id: IRI): AclTargetResource =
```

Comment:
I think the id type of an ACL resource can be the target itself. The toResource won't need a param anymore. I was having the same dilemma with the realms, but since the ResourceF takes an Id type param, setting a different id makes the problem go away.